src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateApprenticeFeedback/CreateApprenticeFeedbackHandler.cs

```diff
@@ -34,11 +34,44 @@ public async Task<CreateApprenticeFeedbackResponse> Handle(CreateApprenticeFeedb
 
             var validAttributes = await _apprenticeFeedbackRepository.GetAttributes();
 
+            var validAttributesNames = validAttributes.Select(attribute => attribute.AttributeName).ToList();
```

Comment:
You are definitely on the right lines here, and what you've got works. but comparing strings in this way can cause issues if for example the strings being compared are different cases eg "WORKING" and "working", though there are ways to mitigate this using StringComparison Flags.

I would stick with your List/Except method but I would base it on the `Id` field instead. Which is the unique key of a given attribute, similiar to the way the `All` method works.

So I would rework the flow to be something like.
- Select All Id's of provided attributes,
- Select all Id's of valid attributes
- Run the Provided ids Except Valid Ids
- If that resulting list is empty, the request is valid
- if that resulting list is not empty, use those id's to retrieve the names from the request to fill in the invalid attributes in your request.

And finally, just for ease of reading, the handler itself is starting to get quite long, how would you re-structure it to try to make it more readable?